### PR TITLE
Update client-go and k8s packages

### PR DESCRIPTION
**Pull request by jaydeokar**
**What type of PR is this?**
cleanup

**Which issue does this PR fix**:
https://github.com/aws/amazon-vpc-cni-k8s/issues/2195

**What does this PR do / Why do we need it**:
Update package versions for client-go and other k8s packages. Also upgrades packages which has security vulnerabilities 

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:
N/A

**Testing done on this change**:
Yes, tested cni, ipamd and metrics-helper 

**Automation added to e2e**:
N/A

**Will this PR introduce any new dependencies?**:
N/A

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
N/A

**Does this change require updates to the CNI daemonset config files to work?**:
N/A

**Does this PR introduce any user-facing change?**:
N/A

```release-note
Update client-go and other k8s package versions
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cmd/egress-v4-cni-plugin/cni.go [hunk 1]
```diff
@@ -120,8 +120,12 @@ func setupContainerVeth(netns ns.NetNS, ifName string, mtu int, pr *current.Resu
 	hostInterface := &current.Interface{}
 	containerInterface := &current.Interface{}
 
+	// We don't specify the container MAC address and depend on linux to define the address
+	// when veth is created
+	containerMac := ""
+
 	err := netns.Do(func(hostNS ns.NetNS) error {
-		hostVeth, contVeth0, err := ip.SetupVeth(ifName, mtu, hostNS)
+		hostVeth, contVeth0, err := ip.SetupVeth(ifName, mtu, containerMac, hostNS)
```
Comment (jdn5126) [inline cmd/egress-v4-cni-plugin/cni.go:128]: nit: just pass an empty string or nil here instead of defining the variable

**Review by jdn5126 — COMMENTED** [review]
Changes LGTM, thanks! Let's call out the `github.com/containernetworking/cni/pkg/types` change in the PR description and link to the upstream documentation. Let's also add a longer form release note.

File: cmd/egress-v4-cni-plugin/cni.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ import (
 
 	"github.com/containernetworking/cni/pkg/skel"
 	"github.com/containernetworking/cni/pkg/types"
-	"github.com/containernetworking/cni/pkg/types/current"
+	current "github.com/containernetworking/cni/pkg/types/100"
```
Comment (jayanthvn) [inline cmd/egress-v4-cni-plugin/cni.go:27]: currently our [spec version](https://github.com/aws/amazon-vpc-cni-k8s/blob/master/misc/10-aws.conflist#L2) is 0.4.0, maybe we can just refer to types/040?

Comment (jaydeokar) [reply cmd/egress-v4-cni-plugin/cni.go:27]: Ah okay, that's what I was concerned about. Let me update that and check

Comment (jaydeokar) [reply cmd/egress-v4-cni-plugin/cni.go]: Kept the spec version to 0.4.0

File: go.mod [hunk 3]
```diff
@@ -8,158 +8,156 @@ require (
 	github.com/aws/amazon-vpc-resource-controller-k8s v1.0.7
```
Comment (jdn5126) [inline go.mod:8]: While we are updating, I think we need to update this to the latest VPC RC version

**Review by jdn5126 — APPROVED** [review]

File: go.mod (hunk 3 above)
Comment (jdn5126) [reply go.mod:8]: Resolving this as it will be taken care of in issue #2229

**Review by jdn5126 — APPROVED** [review]

**Pull request merged**